From cadae3ab7abf45e61ecae8aac39d97d1f3cbd336 Mon Sep 17 00:00:00 2001
From: Lawrence Nahum <lawrence@greenaddress.it>
Date: Sun, 3 Dec 2017 22:56:09 +0100
Subject: [PATCH] fixup

---
 configure.ac          | 1 +
 evutil_rand.c         | 3 +++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7528d37..3bb2121 100644
--- a/configure.ac
+++ b/configure.ac
@@ -341,6 +341,7 @@ dnl Checks for library functions.
 AC_CHECK_FUNCS([ \
   accept4 \
   arc4random \
+  arc4random_addrandom \
   arc4random_buf \
   eventfd \
   epoll_create1 \
diff --git a/evutil_rand.c b/evutil_rand.c
index 046a14b..3f0bf2c 100644
--- a/evutil_rand.c
+++ b/evutil_rand.c
@@ -191,6 +191,7 @@ evutil_secure_rng_get_bytes(void *buf, size_t n)
 {
 	ev_arc4random_buf(buf, n);
 }
+#ifdef HAVE_ARC4RANDOM_ADDRANDOM
 
 void
 evutil_secure_rng_add_bytes(const char *buf, size_t n)
@@ -199,6 +200,8 @@ evutil_secure_rng_add_bytes(const char *buf, size_t n)
 	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
 }
 
+#endif
+
 void
 evutil_free_secure_rng_globals_(void)
 {
-- 
2.14.3
