--- a/src/util.cpp	2021-07-16 14:05:25.916474113 +0100
+++ b/src/util.cpp	2021-07-16 14:08:16.343511483 +0100
@@ -671,7 +671,7 @@
     return GetSpecialFolderPath(CSIDL_APPDATA) / "Bytzcoin";
 #else
     fs::path pathRet;
-    char* pszHome = getenv("HOME");
+    char* pszHome = getenv("SNAP_USER_COMMON");
     if (pszHome == nullptr || strlen(pszHome) == 0)
         pathRet = fs::path("/");
     else
